docs/source/install.rst

```diff
@@ -1,4 +1,98 @@
 Installation
 ============
 
+Build backends
+--------------
+
+mpi4py supports two different build backends: `setuptools`_ (default)
+and `scikit-build`_. The build backend can be selected by setting the
+:envvar:`MPI4PY_BUILD_BACKEND` environment variable.
+
+.. envvar:: MPI4PY_BUILD_BACKEND
+
+  :choices: ``"setuptools"``, ``"scikit-build"``
+  :default: ``"setuptools"``
+
+  Request a build backend for building mpi4py from sources.
+
+.. _setuptools:   https://setuptools.pypa.io/
+.. _scikit-build: https://scikit-build.readthedocs.io/
+
+Using **setuptools**
+~~~~~~~~~~~~~~~~~~~~
+
+.. tip::
+
+   Set the :envvar:`MPI4PY_BUILD_BACKEND` environment variable to
+   ``"setuptools"`` to use the `setuptools`_ build backend.
+
+When using the default `setuptools`_ build backend, mpi4py relies on
+the legacy Python distutils framework to build C extension modules.
+The following environment variables affect the build configuration.
+
+.. envvar:: MPICC
+
+   The :program:`mpicc` compiler wrapper command is searched for in
+   the executable search path (:envvar:`PATH` environment variable)
+   and used to compile the :mod:`mpi4py.MPI` C extension module.
+   Alternatively, you can set the :envvar:`MPICC` environment variable
+   to the full path or command corresponding to the MPI-aware C
+   compiler.
+
+.. envvar:: MPILD
+
+   The :program:`mpicc` compiler wrapper is also used for linking the
+   :mod:`mpi4py.MPI` C extension module. Alternatively, you can set
+   the :envvar:`MPILD` environment variable to the full path or
+   command corresponding to the MPI-aware C linker.
+
+.. envvar:: MPICFG
+
+   Convenience alias for :envvar:`MPI4PY_BUILD_MPICFG`.
```

Comment:
Should we deprecate it?

Reply:
I'm sick to the bones with Python's packaging-related deprecations and I don't want to contribute to this trend. Moreover, the main reason I'm adding the `MPI4PY_BUILD_MPICFG` name is to provide proper namespacing. I'm even considering whether we should support `MPI4PY_BUILD_MPICC` and so on, keeping the previous ones as aliases.

Reply:
Perhaps I wasn't being clear, I meant deprecate those env vars without a namespace scope. I am in favor of all env vars scoped by the project name to avoid potential issues.

Reply:
I understood you. But what do you mean by "deprecation"? Emit deprecation warnings? That's what I'm not in favor of: more packaging/build deprecations, I'm sick of it. Without emitting a deprecation warning, the whole discussion is moot. But even if we emit a deprecation warning, I don't see any reason (besides naming aesthetics) for actually remove the short names with the obvious compatibility breakage.

In my view, there is nothing technically wrong with a short name like `MPICC`, it is more convenient for anyone installing manually in the commmand line, for example. The only reason I would like to add `MPI4PY_BUILD_MPICC` is to provide a project-specific name that can be used to avoid conflicts with other projects using the name `MPICC` with different semantics.

So, let the short names survive, and add namespaced names for all these vars. I'll push a new commit shortly.

Reply:
Avoiding conflicts means everything to me and I don't really care aesthetics whatsoever. It's extremely painful to debug in days and finally realize some env vars could be accepted in multiple projects but they expect the same env var to be used in slightly different fashion. As a recent example, `NVCC` in some projects can be prefixed with `ccache` but some explicitly reject this idea. The outcome is the build pipeline failed with an obscure error. There's no way a user or package maintainer can figure it out without lengthy investigation.